### PR TITLE
Halloween Week 2: Ghosts in the Geiger Counter

### PR DESCRIPTION
**Pull request by CharlesWedge**
## You should probably equip a radsuit.
Adds 4 new spooky creatures, 2 of which emit radiation passively creating a new type of hazard around dealing with them. A short description of the new spooks are as follows:

_**Gammawraith:**_ Resembling a specter wearing a radiation suit. The gamma wraith passively emits a small amount of radiation while firing armor-piercing gamma beams that irradiate the target.
_**Charred Runner:**_ A burnt corpse that runs at its enemies, a glass canon. Very resistant to energy weapons.
**_Burning Runner:_** A extremely fast enemy resembling a burning corpse whose attack sets targets. Almost immune to burn weapons.
**_Demoncore:_** Inflicts death by floating slowly at its enemies and giving them radiation poisoning.

## Enemy Variety
The new mobs provided at least some variety to the existing spooks being distinctively not blood cult themed monsters. In addition, many are dangerous in ways other hostiles simply aren't. The demon core emits radiation which can quickly kill the unprepared (slowly and agonizingly), while the gamma wraith can chip through any armor. The burning and charred also use energy weapon resistance to force melee or bullet combat. The burning specifically is meant to cause chaos by lighting those caught by it on fire (and is fast enough its very hard to kite).

## Changelog

:cl:
add: Four new spooks, ready for Halloween
/:cl:

### CODE REVIEW

**Review by silicons — DISMISSED** [review]

File: code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/gammawraith.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+/mob/living/simple_mob/construct/nuclear/gammawraith/death()
+	..()
+	for(var/mob/M in viewers(src, null))
+		if((M.client && !( M.has_status_effect(/datum/status_effect/sight/blindness) )))
+			M.show_message("<font color='red'>[src] screeches and explodes in a green flash.</font>")
+	playsound(src, 'sound/items/geiger/ext1.ogg', 100, 1)
+	ghostize()
+	qdel(src)
+	return
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/gammawraith.dm:63]: don't need this return

File: code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/demoncore.dm [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	attacktext = list("rammed")
+	attack_sound = 'sound/weapons/rapidslice.ogg'
+	movement_cooldown = 4
+	catalogue_data = list(/datum/category_item/catalogue/fauna/nuclear_spirits/demoncore)
+
+	ai_holder_type = /datum/ai_holder/polaris/simple_mob/melee/evasive
+
+/mob/living/simple_mob/construct/nuclear/demoncore/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/horror_aura/)
```
Comment (silicons) [inline code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/demoncore.dm:32]: unnecessary extra / at the end

File: code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/burning.dm [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	if(M.fire_stacks <= 2)
+		M.adjust_fire_stacks(1)
+		M.IgniteMob()
+
+/mob/living/simple_mob/vore/solarmoth/BiologicalLife(seconds, times_fired)
+	if((. = ..()))
+		return
+
+	if(stat != DEAD)
+		var/datum/gas_mixture/env = loc.return_air() //Gets all the information on the local air.
```
Comment (TheLordME) [inline code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/burning.dm:62]: this returns the air for the turf, not the whole room (each tile in the room has the same mixture as this one)

File: code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/burning.dm [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+			heat_transfer = min(heat_transfer , heating_power) //limit by the power rating of the heater
+			removed.adjust_thermal_energy(heat_transfer)
+
+		else if(heat_transfer > 0 && env.temperature < set_temperature) //Set temperature is 250 degrees celsius. Heating rate should increase between 200 and 450 C.
+			heating_power = original_temp*100
+			heat_transfer = min(heat_transfer , heating_power) //limit by the power rating of the heater. Except it's hot, so yeah.
+			removed.adjust_thermal_energy(heat_transfer)
+
+		else
+			return
```
Comment (TheLordME) [inline code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/burning.dm:76]: you should not return early without merging removed back into env, or you are creating a vacuum once you are at temperature.

File: code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/burning.dm [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+	if(. == 0 && !is_dead())
+		set_light(2.5, 1, COLOR_ORANGE)
+		return 1
+
+/mob/living/simple_mob/construct/nuclear/burning/apply_melee_effects(mob/living/carbon/M, alien, removed)
+	if(M.fire_stacks <= 2)
+		M.adjust_fire_stacks(1)
+		M.IgniteMob()
+
+/mob/living/simple_mob/vore/solarmoth/BiologicalLife(seconds, times_fired)
```
Comment (TheLordME) [inline code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/burning.dm:57]: the comments are misleading and should be revisited

File: code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/demoncore.dm [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+	desc = "Floating in the sky appears to a plutonium sphere undergoing a perpetual criticality incident. It seems to desire to meet you closely."
+	icon_state = "demoncore"
+	icon_living = "demoncore"
+	maxHealth = 200 //Tanky but not tough this thing just wants to be alive long enough to spread its radiation.
+	health = 200
+	response_harm = "viciously beaten"
+	harm_intent_damage = 5
+	legacy_melee_damage_lower = 8 //It's not the strongest of the bunch, but that doesn't mean it can't hurt you.
+	legacy_melee_damage_upper = 10
+	attack_armor_pen = 50 // Does so little damage already, that this can be justified.
```
Comment (TheLordME) [inline code/modules/mob/living/simple_mob/subtypes/occult/constructs/nuclear/demoncore.dm:22]: is 8 to 10 really little damage these days?

**Review by TheLordME — CHANGES_REQUESTED** [review]

**Review by TheLordME — APPROVED** [review]

**Review by TheLordME — APPROVED** [review]

**Pull request merged**